ocaml/typing/persistent_env.ml

```diff
@@ -272,49 +297,109 @@ let process_pers_struct penv check modname pers_sig =
   | true, true
   | false, false -> ()
   end;
-  ps
-
-let bind_pers_struct penv modname ps pm =
-  let {persistent_structures; _} = penv in
-  Hashtbl.add persistent_structures modname (Found (ps, pm))
-
-let acknowledge_pers_struct penv check modname pers_sig pm =
-  let ps = process_pers_struct penv check modname pers_sig in
-  bind_pers_struct penv modname ps pm;
-  ps
-
-let read_pers_struct penv val_of_pers_sig check modname filename ~add_binding =
+  let impl =
+    match kind with
+    | Normal { cmi_impl } -> Impl.Known cmi_impl
+    | Parameter -> Impl.Unknown_argument
+  in
+  let {imports;} = penv in
+  let import = { imp_is_param = is_param;
+                 imp_impl = impl;
+                 imp_sign = sign;
+                 imp_filename = filename;
+                 imp_visibility = visibility;
+                 imp_crcs = crcs;
+                 imp_flags = flags;
+               } in
+  if check then check_consistency penv import;
+  Hashtbl.add imports modname (Found import);
+  import
+
+let read_import penv ~check modname filename =
   add_import penv modname;
```

Comment:
I suspect that `add_import` should be called on a `pers_struct` rather than an `import`. It is used to make sure that all expected compilation units are linked, and if you refer to `Foo(Bar)` it is probably the result of `Foo(Bar)` that you want to make sure is linked, rather than just `Foo`.

Reply:
Is it used for that? I can only find uses relating to CRC checking of interfaces, which is certainly something that makes sense at the level of `foo.cmi` and `bar.cmi`.

Reply:
Ah, I was confusing it with `Env.required_globals`